src/transforms/sample/config.rs

```diff
@@ -44,6 +44,11 @@ pub struct SampleConfig {
     #[configurable(metadata(docs::examples = "message"))]
     pub key_field: Option<String>,
 
+    /// A field for grouping sampling by a particular log field. Logs that come in with a unique
+    /// value for the field will be sampled together.
+    #[configurable(metadata(docs::examples = ".log_field"))]
+    pub group_by: Option<String>,
```

Comment:
```suggestion
    pub group_by: Option<Template>,
```

Could we make this a template field? That way people could join together multiple fields to create the groupings (e.g. they could do things like `{{ host }}-{{ service }}`. See the `throttle` transform for an example of how the `group_by` would be rendered: https://github.com/vectordotdev/vector/blob/a1edf90e9b78ace1cbb0b579324b1d5b027754b5/src/transforms/throttle.rs#L55-L60